### DISCUSSION TITLE
Intercept a $patch

### DISCUSSION THREAD

**Discussion by ghost23** [vuejs/pinia#2632] [Help and Questions] [answered]
Hello,

In my store, I would like to intercept $patch(..) calls and do some validation first before the store is actually updated. Is this possible?

**Comment by posva** [comment, marked as answer]
You cannot _intercept_, only react to patches with `store.$subscribe()` (can be within a pinia plugin too). But it's not possible to do any validation. It sounds that what you want to do is an action that handles the validation of the input. It would help to add [private state](https://masteringpinia.com/blog/how-to-create-private-state-in-stores) to that

**Reply by ghost23** [reply]
OK, thank you.
